server.go

```diff
@@ -490,17 +1276,37 @@ func (s *server) handleQuery(querymsg interface{}, state *peerState) {
 		}
 
 		// TODO(oga) if too many, nuke a non-perm peer.
-		if s.handleAddPeerMsg(state,
-			newOutboundPeer(s, msg.addr, msg.permanent, 0)) {
+		peerCfg := &peer.Config{
```

Comment:
This should be defined after the connection.  No reason to create it when the connection fails (which is quite often).  Once you've expanded `peer.Config` to include the connection, it would be more clear anyways.